sdl2promela/translator.py

```diff
@@ -212,7 +215,10 @@ def __is_local_variable(context: Context, variable: str):
             for parameter in procedure.parameters:
                 if parameter.name == variable:
                     return True
-    return False
+    if variable.lower() in context.sdl_model.variables:
+        return False
```

Comment:
Why?

Reply:
The iterator variable in `for` is a local variable, to detect this I added this check.